src/components/PaymentFinalizationPage/PaymentFinalizationPage.js

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { StyledOfferWrapper } from 'components/Offer/OfferStyled';
+import Header from 'components/Header';
+import Footer from 'components/Footer';
+import { fetchFinalizeInitialPayment } from 'redux/finalizePaymentSlice';
+import { useDispatch, useSelector } from 'react-redux';
+import { StyledLoaderContent } from 'containers/OfferContainer/StyledOfferContainer';
+import Loader from 'components/Loader';
+import ThankYouPage from 'components/ThankYouPage';
+import FailedPaymentPage from 'components/FailedPaymentPage';
+
+const PaymentFinalizationPage = ({ onSuccess }) => {
+  const adyenRedirectResult = new URLSearchParams(window.location.search).get(
+    'redirectResult'
+  );
+  const orderId = new URLSearchParams(window.location.search).get('orderId');
+  const dispatch = useDispatch();
+
+  const submitRedirectResult = async () => {
```

Comment:
do we need async here?

Reply:
that's true. I will remove it also from other similar places